File: src/ragent/domain/enums.py
"""领域枚举。

纯领域，零业务依赖。供 persistence / service / schemas 等层共享。
"""

from __future__ import annotations

from enum import StrEnum


class DocumentStatus(StrEnum):
    """文档状态机：pending → parsing → chunking → embedding → indexing → completed/failed。"""

    PENDING = "pending"
    PARSING = "parsing"
    CHUNKING = "chunking"
    EMBEDDING = "embedding"
    INDEXING = "indexing"
    COMPLETED = "completed"
    FAILED = "failed"


class ChunkEmbeddingStatus(StrEnum):
    """分块 Embedding 状态。"""

    PENDING = "pending"
    EMBEDDED = "embedded"
    INDEXED = "indexed"
    FAILED = "failed"


class MessageRole(StrEnum):
    """对话消息角色。"""

    USER = "user"
    ASSISTANT = "assistant"


class ProviderType(StrEnum):
    """模型供应商类型。"""

    QWEN = "qwen"
    OPENAI = "openai"
    DEEPSEEK = "deepseek"


class KnowledgeBaseStatus(StrEnum):
    """知识库状态。"""

    ACTIVE = "active"
    ARCHIVED = "archived"

File: web/src/types/enums.ts
/**
 * 枚举常量（与后端 src/ragent/domain/enums.py 值对齐）
 * 仅作前端常量，不依赖后端代码
 */

/** 文档状态机 */
export const DocumentStatus = {
  PENDING: 'pending',
  PARSING: 'parsing',
  CHUNKING: 'chunking',
  EMBEDDING: 'embedding',
  INDEXING: 'indexing',
  COMPLETED: 'completed',
  FAILED: 'failed',
} as const

export type DocumentStatusValue = (typeof DocumentStatus)[keyof typeof DocumentStatus]

/** 知识库状态 */
export const KnowledgeBaseStatus = {
  ACTIVE: 'active',
  ARCHIVED: 'archived',
} as const

export type KnowledgeBaseStatusValue =
  (typeof KnowledgeBaseStatus)[keyof typeof KnowledgeBaseStatus]
